### PR TITLE
[3.9][com_admin] warning in Edit Account

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #31610 .

### Summary of Changes
added  key exist check

### Testing Instructions
Go to System - Control Panel.
Click Super User – Edit Account.
Go to any tab, for example User Profile and edit something, then click Save.
Or do not change anything and click Save.

### Actual result BEFORE applying this Pull Request
Message displayed: 0 Cannot access offset of type string on string

### Expected result AFTER applying this Pull Request

No warnings or errors.

### CODE REVIEW

**Comment by ghost** [comment]
@SharkyKZ why thumbs down?

**Comment by toivo** [comment]
@alikon PHP 8.0.0 reports the variable $data<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31628">issues.joomla.org/tracker/joomla-cms/31628</a>.</sub>

**Comment by toivo** [comment]
@alikon PHP 8.0.0 reports that the variable $data is not an array, even though var_dump($data) shows that it is an array.  This is probably some teething problem in an early version of PHP 8.

The following modification allowed the profile data to be saved correctly:
```
	if (is_array($data) && array_key_exists('twofactor', (array) $data) && array_key_exists('method', (array) $data['twofactor']))
```
I have not tested any profile with two-factor authentication configured.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31628">issues.joomla.org/tracker/joomla-cms/31628</a>.</sub>

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 98872a15937bfe1cb5c7abba9dc1470d0b6c2f7d<br /><br />Tested successfully in 3.9.23.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31628">issues.joomla.org/tracker/joomla-cms/31628</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 98872a15937bfe1cb5c7abba9dc1470d0b6c2f7d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31628">issues.joomla.org/tracker/joomla-cms/31628</a>.</sub>

**Comment by jwaisner** [comment]
Drone Error is not related to PR.

RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31628">issues.joomla.org/tracker/joomla-cms/31628</a>.</sub>

**Comment by infograf768** [comment]
restarted drone

File: administrator/components/com_admin/models/profile.php [hunk 1]
```diff
@@ -140,7 +140,7 @@ public function save($data)
 		}
 
 		// Handle the two factor authentication setup
-		if (array_key_exists('twofactor', $data))
+		if ((array_key_exists('twofactor', $data)) && (isset($data['twofactor']['method'])))
```
Comment (HLeithner) [inline administrator/components/com_admin/models/profile.php]:
```suggestion
		if (isset($data['twofactor']['method']))
```

**Comment by HLeithner** [comment]
@alikon please update your pr and both tests  (@toivo and @gostn ) are invalid and have to be redone (after changing the PR)

**Comment by jwaisner** [comment]
Moving back to Pending for PR changes. Clearing test results as the changes will require retest.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31628">issues.joomla.org/tracker/joomla-cms/31628</a>.</sub>

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 7b4ab73f9c7ce286ea6033d2c432960046e46500<br /><br />Tested successfully in 3.9.24-dev of 13 December.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31628">issues.joomla.org/tracker/joomla-cms/31628</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 7b4ab73f9c7ce286ea6033d2c432960046e46500<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31628">issues.joomla.org/tracker/joomla-cms/31628</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31628">issues.joomla.org/tracker/joomla-cms/31628</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Comment by PhilETaylor** [comment]
Tested 3.9.24RC. All ok.

**Pull request merged**